src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.cc

```diff
@@ -202,10 +203,15 @@ void FileWatcherCertificateProvider::ForceUpdate() {
       (pem_key_cert_pairs.has_value() &&
        pem_key_cert_pairs_ != *pem_key_cert_pairs);
   if (identity_cert_changed) {
-    if (pem_key_cert_pairs.has_value()) {
-      pem_key_cert_pairs_ = std::move(*pem_key_cert_pairs);
+    absl::Status key_cert_match = PrivateKeyAndCertificateMatch(primary_key_path_, identity_certificate_path_);
```

Comment:
This doesn't look right. We will need to pass the credential contents, instead of the file path.

Reply:
ok, fixing that